src/ProjectTemplates/test/RazorComponentsTemplateTest.cs

```diff
@@ -75,35 +75,31 @@ private void TestBasicNavigation()
             // Give components.server enough time to load so that it can replace
             // the prerendered content before we start making assertions.
             Thread.Sleep(5000);
-            Browser.WaitForElement("ul");
+            Browser.Exists(By.TagName("ul"));
             // <title> element gets project ID injected into it during template execution
-            Assert.Contains(Project.ProjectGuid, Browser.Title);
+            Browser.Equal(Project.ProjectName.Trim(), () => Browser.Title.Trim());
 
             // Initially displays the home page
-            Assert.Equal("Hello, world!", Browser.GetText("h1"));
+            Browser.Equal("Hello, world!", () => Browser.FindElement(By.TagName("h1")).Text);
 
             // Can navigate to the counter page
-            Browser.Click(By.PartialLinkText("Counter"));
-            Browser.WaitForUrl("counter");
-            Browser.WaitForText("h1", "Counter");
+            Browser.FindElement(By.PartialLinkText("Counter")).Click();
+            Browser.Contains("counter", () => Browser.Url);
+            Browser.Equal("Counter", () => Browser.FindElement(By.TagName("h1")).Text);
 
             // Clicking the counter button works
-            var counterComponent = Browser.FindElement("h1").Parent();
-            var counterDisplay = Browser.FindElement("h1 + p");
```

Comment:
I see you didn't want to hang onto the same element references, but to save me from making mistakes in the future, could you explain why this is wrong to do?

I would have reasoned that, by the time we are on the "Counter" page, the `counterDisplay` element instance is going to remain fixed, so it's safe to hold it. Arguably it's even preferable to do so, as it shows we're not doing completely broken things like tearing down the entire UI and recreating it after each click.

Reply:
I didn't think much of that. I was trying to make the tests terser

Reply:
OK, sounds fine.